### PR TITLE
Verify mount point after erase

### PR DESCRIPTION
**Pull request by juhhov**
## Status
**IN DEVELOPMENT**

## Description
Wait for device mount point to reappear after erase before returning.
Also check Daplink version before starting erase operation, fail if Daplink version is too old.

## TODO

- [x] tests

### CODE REVIEW

File: mbed_flasher/erase.py [hunk 1]
```diff
@@ -98,61 +100,113 @@ def reset_board(self, serial_port):
...
+        :param mount_point: mount_point to watch disappear
+        """
+        start_time = time()
+        while True:
+            sleep(0.1)
+            if platform.system() == 'Windows':
+                proc = Popen(["dir", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE, shell=True)
+                err = proc.stderr.read()
+            else:
+                proc = Popen(["ls", mount_point], stdin=PIPE, stdout=PIPE, stderr=PIPE)
```
Comment (jonikula) [inline mbed_flasher/erase.py]: Why is the shell-param not used here?

File: mbed_flasher/erase.py [hunk 2]
```diff
@@ -98,61 +100,113 @@ def reset_board(self, serial_port):
...
-    def runner(self, drive):
+    def wait_to_disappear(self, mount_point):
         """
-        :param drive:
+        :param mount_point: mount_point to watch disappear
+        """
+        start_time = time()
+        while True:
+            sleep(0.1)
+            if platform.system() == 'Windows':
```
Comment (jonikula) [inline mbed_flasher/erase.py]: Wouldn't it be better to determine the correct command outside the loop?

File: mbed_flasher/common.py [hunk 3]
```diff
@@ -32,3 +37,131 @@ def __init__(self, name):
...
+                                  target['serial_port'])
+            if 'mount_point' in new_target:
+                self.logger.debug("mount point %s has changed to %s",
+                                  target['mount_point'], new_target['mount_point'])
+            else:
+                self.logger.debug("mount point %s has not changed",
+                                  target['mount_point'])
+            new_target['target_id'] = target['target_id']
+            return new_target
+        else:
```
Comment (jonikula) [inline mbed_flasher/common.py]: Else not needed, just the return is enough.

File: mbed_flasher/erase.py (hunk 1 above)
Comment (juhhov) [reply mbed_flasher/erase.py]: Shell option is not needed for linux command. In windows shell=True directs the command to cmd.exe which we want to use.

Comment (jonikula) [reply mbed_flasher/erase.py]: Is there a downside to using shell=True in linux? That way you could easily just select the started process array based on system outside this loop.

Comment (juhhov) [reply mbed_flasher/erase.py]: https://docs.python.org/2/library/subprocess.html?highlight=subprocess%20popen#popen-constructor
The box with red content.

File: mbed_flasher/common.py [hunk 4]
```diff
@@ -32,3 +37,131 @@ def __init__(self, name):
...
+        return target
+
+    def _check_serial_point_duplicates(self, target, new_target):
+        """
+        Verify that target is not listed multiple times in /dev/serial/by-id
+        :param target: old target
+        :param new_target: new target
+        :return: if all is well None, otherwise error code
+        """
+        for line in os.popen('ls -oA /dev/serial/by-id/').read().splitlines():
```
Comment (YifeiZuo01) [inline mbed_flasher/common.py]: What about this Popen  without close?

File: mbed_flasher/common.py [hunk 5]
```diff
@@ -153,7 +155,7 @@ def _verify_mount_point(self, target, new_target):
             return -12
 
         for _ in range(MountVerifier.MOUNT_POINT_TIMEOUT):
-            mounts = os.popen('mount |grep vfat').read().splitlines()
+            mounts = check_output(["mount", "|", "grep", "vfat"], shell=True).splitlines()
```
Comment (YifeiZuo01) [inline mbed_flasher/common.py]: then, why this one add shell=True? This is only for linux usage?

Comment (juhhov) [reply mbed_flasher/common.py]: Pipe is a shell feature, it won't work without. Here input is always the same (secure) anyways.

Comment (juhhov) [reply mbed_flasher/common.py]: Could do the filtering in python however, maybe some other PR.

**Comment by YifeiZuo01** [comment]
please fix the lint warnings.

**Pull request merged**